src/api/firebase.js

```diff
@@ -160,3 +160,45 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(data) {
+	const today = new Date();
+	const newData = data.map((item) => ({
+		...item,
+		purchaseUrgency: purchaseSchedule(
+			item.dateLastPurchased,
+			item.dateNextPurchased,
+		),
+		daysTillNextPurchase: getDaysBetweenDates(
+			today,
+			item.dateNextPurchased.toDate(),
+		),
+	}));
+	const compareFn = (a, b) => {
```

Comment:
Would this function be better off declared outside of `comparePurchaseUrgency`, i.e not nested? 
nit: the naming on this could be better. `compareFn` doesn't say much. Is it sorting the items by date? So maybe something along the lines of `sortItemsByDate(item1, item2)`?

Reply:
That's a good point, we've renamed this function to `sortItemsByDate` and moved it to `utils/date.js`

Reply:
Thanks @mentalcaries for all the suggestions. We implemented your comments and resolved the git conflict as well!